packages/ember-metal/lib/meta.js

```diff
@@ -354,63 +365,120 @@ if (isEnabled('mandatory-setter')) {
   };
 }
 
+const HAS_NATIVE_WEAKMAP = (function() {
+  // detect if `WeakMap` is even present
+  let hasWeakMap = typeof WeakMap === 'function';
+  if (!hasWeakMap) { return false; }
+
+  let instance = new WeakMap();
+  // use `Object`'s `.toString` directly to prevent us from detecting
+  // polyfills as native weakmaps
+  return Object.prototype.toString.call(instance) === '[object WeakMap]';
```

Comment:
Looks like lodash has been through this one already:

* https://github.com/lodash/lodash/blob/4.15.0/dist/lodash.js#L11583-L11589
* https://github.com/lodash/lodash/blob/4.15.0/dist/lodash.js#L3277-L3291
* https://github.com/lodash/lodash/blob/4.15.0/dist/lodash.js#L6089-L6098
* https://github.com/lodash/lodash/blob/4.15.0/dist/lodash.js#L1427-L1431

Quoting (just before `_.isNative` throws when it detects core-js):

     * **Note:** This method can't reliably detect native functions in the presence
     * of the core-js package because core-js circumvents this kind of detection.
     * Despite multiple requests, the core-js maintainer has made it clear: any
     * attempt to fix the detection will be obstructed. As a result, we're left
     * with little choice but to throw an error. Unfortunately, this also affects
     * packages, like [babel-polyfill](https://www.npmjs.com/package/babel-polyfill),
     * which rely on core-js.